app/src/App.js

```diff
@@ -48,76 +50,77 @@ const App = () => {
     return <div>Loading ...</div>;
   }
 
-  const username = user.email ? user.email : user.username;
+  if (isAuthenticated) {
+    const username = user.email ? user.email : user.username;
+    return (
+      <Router>
+        <main>
+          <Container fluid>
```

Comment:
If you're going to use `<main>`, it'd be a good idea to wrap the Navbar in `<header>` or `<nav></nav>`. 

I think generally the `<main>` tag refers to the body of the page, so you'd move it down between lines `100`  and `101`,

```js
     ...
   </Container>
<header>
<main>
   <Container>
     ...
```

Reply:
It might also be worthwhile idea putting lines `59 - 99` into it's own file called `NavBar` or `Header` or something -- the idea of "separation of concerns", so it's easier to debug and read too.